pkg/kubelet/cm/container_manager_linux.go

```diff
@@ -965,6 +965,35 @@ func (cm *containerManagerImpl) GetAllocatableMemory() []*podresourcesapi.Contai
 	return containerMemoryFromBlock(cm.memoryManager.GetAllocatableMemory())
 }
 
+func (cm *containerManagerImpl) GetDynamicResources(pod *v1.Pod, container *v1.Container) []*podresourcesapi.DynamicResource {
+	if !utilfeature.DefaultFeatureGate.Enabled(kubefeatures.DynamicResourceAllocation) {
+		return []*podresourcesapi.DynamicResource{}
+	}
+	if !utilfeature.DefaultFeatureGate.Enabled(kubefeatures.KubeletPodResourcesDynamicResources) {
+		return []*podresourcesapi.DynamicResource{}
+	}
```

Comment:
I'm not sure we need this extra check here. We should check this FG only at podresources layer, it seems weird to run this check also in the container manager layer. If the podresources api FG is not enabled this function should never be called in the first place.
If you want to be super careful, perhaps add a log on line 970 just before the return? I'd say V(3)

Reply:
It not extra check. It is to avoid doing it on the podresources layer. Do you think it better to move it there [1] ?
We will need to this on List and Get and because the call is per container. 

[1] - https://github.com/kubernetes/kubernetes/pull/115847/files#diff-4240391207d84af10161aab4338a77e780479cf90c5565d5b3f8817fbdd5efb5R129

Reply:
Yes, please move it up and check it in both places.